keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -0,0 +1,789 @@
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+### Exporting Services
+
+Services will not be visible to other clusters in the supercluster by default.
+They must be explicitly marked for export by the user. This allows users to
+decide exactly which services should be visible outside of the local cluster.
+
+Tooling may (and likely will, in the future) be built on top of this to simplify
+the user experience. Some initial ideas are to allow users to specify that all
+services in a given namespace or in a namespace selector or even a whole cluster
+should be automatically exported by default. In that case, a `ServiceExport`
+could be automatically created for all `Services`. This tooling will be designed
+in a separate doc, and is secondary to the main API proposed here.
+
+To mark a service for export to the supercluster, a user will create a
+ServiceExport CR:
+
+```golang
+// ServiceExport declares that the associated service should be exported to
+// other clusters.
+type ServiceExport struct {
+        metav1.TypeMeta `json:",inline"`
+        // +optional
+        metav1.ObjectMeta `json:"metadata,omitempty"`
+        // +optional
+        Status ServiceExportStatus `json:"status,omitempty"`
+}
+
+// ServiceExportStatus contains the current status of an export.
+type ServiceExportStatus struct {
+        // +optional
+        Conditions []ServiceExportCondition `json:"conditions,omitempty"`
+}
+
+type ServiceExportConditionType string
+
+const {
+      // ServiceExportInitialized means the service export has been noticed
+      // by the controller, has passed validation, has appropriate finalizers
+      // set, and any required supercluster resources like the IP have been
+      // reserved
+      ServiceExportInitialized ServiceExportConditionType = "Initialized"
+      // ServiceExportExported means that the service referenced by this
+      // service export has been synced to all clusters in the supercluster
+      ServiceExportExported ServiceExportConditionType = "Exported"
+}
+
+// ServiceExportCondition contains details for the current condition of this
+// service export.
+type ServiceExportCondition struct {
```

Comment:
FYI: https://github.com/kubernetes/enhancements/pull/1624

Reply:
Assuming it will be merged, we should make it consistent.

Reply:
Agreed - made a note